### PR TITLE
[Merged by Bors] - refactor: move material about the Dominated Convergence Theorem into one file

### PR DESCRIPTION
**Pull request by grunweg**
[Suggested](https://github.com/leanprover-community/mathlib4/pull/11108#issuecomment-1975318101) by @loefflerd.
Only code motion (and cosmetic adaptions, such as minimising import and open statements).

Pre-requisite for #11108 and (morally) #11110.

-----------

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by loefflerd** [comment]
Here's a useful hack for fixing downstream imports when you've moved a lot of code around.

- Make a temporary file in mathlib which has no content, but imports all the other files you've changed in the refactor.
- Make every file elsewhere in mathlib which imported one of the changed files import your new temporary file.
- Let CI run on this, and then do `lake exe shake --fix`.

**Comment by loefflerd** [comment]
FWIW, I'm not sure if there's a standard convention about copyright / author declarations for newly-created files which are made up of code moved from elsewhere in a refactor, but I usually handle this by copying across the author names from the previous files.

**Comment by grunweg** [comment]
Thanks for the suggestions. These are really helpful, going forward --- my current approach was a bit tedious.

The imports were basically already fixed, so I didn't need your approach this time. For the copyright, used git blame (on mathlib :-(, the history merge would have been useful) to find the main authors of the respective lemmas.

I have merged master, completed the module docstring and fixed the copyright. This is ready for review again.

File: Mathlib/MeasureTheory/Integral/DominatedConvergence.lean [hunk 1]
```diff
@@ -0,0 +1,518 @@
+/-
+Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou, Yury Kudryashov, Patrick Massot
+-/
+import Mathlib.MeasureTheory.Integral.IntervalIntegral
+import Mathlib.Order.Filter.IndicatorFunction
+
+/-!
+## The dominated convergence theorem
```
Comment (loefflerd) [inline Mathlib/MeasureTheory/Integral/DominatedConvergence.lean:10]:
```suggestion
# The dominated convergence theorem
```

File: Mathlib/MeasureTheory/Integral/DominatedConvergence.lean [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+for the Bochner integral.
+
+## Main results
+- `tendsto_integral_of_dominated_convergence`: the Lebesgue dominated convergence theorem
+- `hasSum_integral_of_dominated_convergence`: the Lebesgue dominated convergence theorem for series
+- `integral_tsum`, `integral_tsum_of_summable_integral_norm`: the integral and `tsum`s commute,
+  if the norms of the functions form a summable series
+- `intervalIntegral.hasSum_integral_of_dominated_convergence`: the Lebesgue dominated convergence
+  theorem for parametric interval integrals
+- `continuous_of_dominated_interval`: continuity of the interval integral w.r.t. a parameter
```
Comment (loefflerd) [inline Mathlib/MeasureTheory/Integral/DominatedConvergence.lean:22]:
```suggestion
- `intervalIntegral.continuous_of_dominated_interval`: continuity of the interval integral w.r.t. a parameter
```
(always give fully qualified names in the file-level summary). Please check the other lemma names too.

**Review by loefflerd — COMMENTED** [review]
Looks great! I left a couple of minor comments on the file docstring.

**Comment by grunweg** [comment]
awaiting-review

**Review by loefflerd — APPROVED** [review]
LGTM!

**Comment by loefflerd** [comment]
maintainer merge

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**